src/main/java/org/motechproject/mots/mapper/ModuleAssignmentMapper.java

```diff
@@ -41,24 +48,26 @@ default CommunityHealthWorker toChw(String chwId) {
    * @param moduleId id of Module.
    * @return Module with given id
    */
-  default Module toModule(String moduleId) {
+  protected Module toModule(String moduleId) {
     if (moduleId == null || moduleId.isEmpty()) {
       return null;
     }
 
     return new Module(UUID.fromString(moduleId));
   }
 
-  /**
-   * Get module id as String.
-   * @param module module assigned to CHW
-   * @return module id as String
-   */
-  default String fromModule(Module module) {
-    if (module == null || module.getId() == null) {
-      return null;
+  @AfterMapping
+  protected void fillIsStaredAndRemoveName(AssignedModules assignedModules,
```

Comment:
Remove this and add method "ModuleSimpleDto toDto(Module module)" instead, you will not need to iterate through modules, you can add ignore on name there, also add fillIsStared(Module module, MappingTarget ModuleSimpleDto moduleDto)"